### PR TITLE
add dummy-import-rgx option

### PR DESCRIPTION
**Pull request by IamRezaMousavi**
## Summary
check #4250

### CODE REVIEW

File: crates/ruff/src/checkers/ast/mod.rs [hunk 1]
```diff
@@ -5135,6 +5135,10 @@ impl<'a> Checker<'a> {
                         _ => continue,
                     };
 
+                    if self.settings.dummy_import_rgx.is_match(full_name) {
+                        continue;
+                    }
+
```
Comment (MichaReiser) [inline crates/ruff/src/checkers/ast/mod.rs:5141]: Nit: I recommend moving the line below the line 5152. The `ImportFrom` check should be cheaper than testing the name.

File: crates/ruff/src/checkers/ast/mod.rs [hunk 2]
```diff
@@ -5135,6 +5135,10 @@ impl<'a> Checker<'a> {
...
                     let child_id = binding.source.unwrap();
                     let parent_id = self.semantic_model.stmts.parent_id(child_id);
```
Comment (MichaReiser) [inline crates/ruff/src/checkers/ast/mod.rs:5143]: Do we need to add an entry to the `ignored` vec @charliermarsh ?

File: crates/ruff/src/checkers/ast/mod.rs (hunk 1 above)
Comment (IamRezaMousavi) [reply crates/ruff/src/checkers/ast/mod.rs:5141]: This's a better idea

File: crates/ruff/src/settings/defaults.rs [hunk 3]
```diff
@@ -32,6 +32,9 @@ pub const TASK_TAGS: &[&str] = &["TODO", "FIXME", "XXX"];
 pub static DUMMY_VARIABLE_RGX: Lazy<Regex> =
     Lazy::new(|| Regex::new("^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$").unwrap());
 
+pub static DUMMY_IMPORT_RGX: Lazy<Regex> =
+    Lazy::new(|| Regex::new("^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$").unwrap());
+
```
[the remark this answers is not in the document]
Comment (IamRezaMousavi) [reply crates/ruff/src/settings/defaults.rs:37]: It's not compatible

File: crates/ruff/src/checkers/ast/mod.rs (hunk 1 above)
Comment (IamRezaMousavi) [reply crates/ruff/src/checkers/ast/mod.rs:5141]: But what if I want import like this?
If I just want to ignore _Name:

```python
from ImportFrom import Name as _Name
```
or
```python
from ImportFrom import OtherName, _Name
```

File: crates/ruff/src/settings/defaults.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (IamRezaMousavi) [reply crates/ruff/src/settings/defaults.rs:37]: It's not compatible with previous version of ruff

[the remark this answers is not in the document]
Comment (IamRezaMousavi) [reply crates/ruff/src/settings/defaults.rs:37]: My mistake, I thought you wanted to change `dummy_variable_rgx` option to `dummy-variable-or-import-rgx`

**Comment by charliermarsh** [comment]
I have two hesitations here that I'm not certain how to address:

1. What rules should this affect? Right now, it only affects `unused_import`. But the setting is highly generic (`dummy-import-rgx`). In theory, should it also effect all other rules that flag imports? Like the wildcard import rules (`from resources import *`)? If not, we need a more deliberate name, like `side_effect_imports` or `effectful_modules` etc.

1. Should this be a regex, or should it instead be a list of module paths? `dummy-variable-rgx` makes sense because it's matching on arbitrary identifiers. But here, we're matching against specific modules, like `foo.bar.baz`. I would prefer, I think, to require the user to pass in a list of fully-qualified module paths.

**Comment by IamRezaMousavi** [comment]
Can I share an idea?


> 1. What rules should this affect? Right now, it only affects `unused_import`. But the setting is highly generic (`dummy-import-rgx`). In theory, should it also effect all other rules that flag imports? Like the wildcard import rules (`from resources import *`)? If not, we need a more deliberate name, like `side_effect_imports` or `effectful_modules` etc.

I think `dummy-import-rgx` look like `dummy-variable-rgx` which the simple way to ignored all rules about this import name 
ps1: Can we merge this both in one option?
ps2: I think we mustn't delete `dummy-variable-rgx` for compatible reason, but can use that for ignore import name. What you think?

> 2. Should this be a regex, or should it instead be a list of module paths? `dummy-variable-rgx` makes sense because it's matching on arbitrary identifiers. But here, we're matching against specific modules, like `foo.bar.baz`. I would prefer, I think, to require the user to pass in a list of fully-qualified module paths.

This is another option. This option look like `per-file-ignores`, we can implement this and give users the option to decide which import rules to ignore for which package

**Comment by charliermarsh** [comment]
I think I'm going to close this one out for now (in shame, since I never got back to you on your follow-up ideas). My current feeling is that I'm hesitant to add more configuration options without thinking through configuration more holistically, which is something we have on the roadmap. I'm sorry for both (1) not resolving your immediate problem and (2) being be a better maintainer / partner on this one.

**Pull request closed without merging**